tests/foreman/cli/test_contentview.py

```diff
@@ -2201,6 +2204,59 @@ def test_positive_publish_custom_content(self):
             'Publishing new version of CV was not successful',
         )
 
+    @tier2
+    @run_only_on('sat')
+    def test_positive_publish_custom_content_module_stream(self):
+        """attempt to publish a content view containing custom content
+        module streams
+
+        :id: 435e49f0-2891-4b04-98b1-bf303dd3a135
+
+        :setup: Multiple repositories for an org and custom content synced
+
+        :expectedresults: Content view published with module streams
+            and count should be correct each time even after republishing CV
+
+        :CaseLevel: Integration
+        """
+        animal_repo = make_repository({u'product-id': self.product['id'],
+                                       u'content-type': u'yum',
+                                       u'url': CUSTOM_MODULE_STREAM_REPO_1}
+                                      )
+
+        software_repo = make_repository({u'product-id': self.product['id'],
+                                         u'content-type': u'yum',
+                                         u'url': CUSTOM_MODULE_STREAM_REPO_2})
+
+        # Sync REPO's
+        Repository.synchronize({'id': animal_repo['id']})
+        Repository.synchronize({'id': software_repo['id']})
+        # Create CV
+        new_cv = make_content_view({u'organization-id': self.org['id']})
+        # Associate repo to CV
+        ContentView.add_repository({
+            u'id': new_cv['id'],
+            u'repository-id': animal_repo['id'],
+        })
+        # Publish a new version of CV
+        ContentView.publish({u'id': new_cv['id']})
+        new_cv_version_1 = ContentView.info({u'id': new_cv['id']})['versions'][0]
+        module_streams = ModuleStream.list({'content-view-version-id': (new_cv_version_1['id'])})
+        self.assertEqual(
+            len(module_streams), 5,
+            'Module Streams are not associated with Content View')
+        # Publish another new version of CV
+        ContentView.add_repository({
+            u'id': new_cv['id'],
+            u'repository-id': software_repo['id'],
+        })
+        ContentView.publish({u'id': new_cv['id']})
+        new_cv_version_2 = ContentView.info({u'id': new_cv['id']})['versions'][1]
+        module_streams = ModuleStream.list({'content-view-version-id': new_cv_version_2['id']})
+        self.assertEqual(
```

Comment:
@omkarkhatavkar Will there always be 12 modules streams at given url? What about  using `self.assertGreater()`

Reply:
@ntkathole make sense! thanks

Reply:
yes, will these repos stay unchanged for us in the future? If not sure, one possibility is to create your own copy on fedorapeople.org